go/vt/vtgate/executor.go

```diff
@@ -115,6 +115,9 @@ type (
 		AllowScatter        bool
 		WarmingReadsPercent int
 		QueryLogToFile      string
+
+		// DefaultMultiShardAutocommit will opt into autocommit semantics even for multi shard DMLs
+		DefaultMultiShardAutocommit bool
```

Comment:
nit: This is not used by executor, it is only passed to vcusor. We can set on the vcursor config directly.